main.py

```diff
@@ -1 +1,179 @@
-print("hello world ")
\ No newline at end of file
+import random
+
+EMPTY_CELL = "."
+FLOWER = "F"
+WEED = "W"
+
+def plant_random(garden , amount , item , row , col):
+    for i in range(amount):
+        while True : 
+            coordi_flower_x = random.randint(0, row - 1)
+            coordi_flower_y = random.randint(0, col - 1)
+            if garden[coordi_flower_x][coordi_flower_y] == EMPTY_CELL:
+                garden[coordi_flower_x][coordi_flower_y] = item
+                break
+    return garden
+        
+
+
+def create_garden(row, col):
+    """Return a Grid for the Garden with n by m squares and 4 flowers and 3 weedss ."""
+    garden = []
+
+    for r in range(row):  
+        rows = []  
+        for c in range(col):  
+            cell_value = EMPTY_CELL  
+            roulette = random.randint(1, 10)  
+            if roulette == 1:  
+                random_fifty = random.randint(1, 2)  
+                if random_fifty == 1:  
+                    cell_value = FLOWER  
+                elif random_fifty == 2:  
+                    cell_value = WEED  
+            rows.append(cell_value)  
+        garden.append(rows) 
+    
+    plant_random(garden , 4 , FLOWER , row , col)
+    plant_random(garden , 3 ,WEED ,  row , col)
+    
+    return garden
+
+
+# function to display the garden
+
+
+def display_garden(garden, horizontal, vertical):
+    """Return a grid of n by m squares"""
+    for r in range(horizontal):
+        for c in range(vertical):
+            print(garden[r][c], end="")
+        print("")
+
+
+
+
+
+# counts the total number of flowers and weeds
+def counting_items(garden, horizontal, vertical):
+    """Return the number of flowers and weeds in the garden"""
+    flower_count = 0
+    weed_count = 0
+    for rows in range(horizontal):
+        for column in range(vertical):
+            if garden[rows][column] == FLOWER:
+                flower_count += 1
+            elif garden[rows][column] == WEED:
+                weed_count += 1
+    return flower_count, weed_count
+
+
+# Ensures fulfillment of condittions with eachh passing day in the simulation where each flower and weed takes over adjacent squares
+
+def plant(garden ,coordi_spread_x , coordi_spread_y , item):
+    try:
+        if garden[coordi_spread_x ][coordi_spread_y] == EMPTY_CELL:
+           garden[coordi_spread_x][coordi_spread_y] = item
+    except IndexError :
+        pass 
+    return garden
```

Comment:
If this function returns `bool`, instead of garden(list), you can broaden the use of it.